js/main.js

```diff
@@ -20,4 +20,94 @@ $form.addEventListener('submit', function (event) {
   data.entries.unshift(newEntry);
   $entryImage.setAttribute('src', 'images/placeholder-image-square.jpg');
   $form.reset();
+  $list.prepend(renderEntry(newEntry));
+  viewSwap('entries');
+  if (data.entries !== null) {
+    toggleNoEntries();
+  } else {
+    toggleNoEntries();
+  }
+});
+
+function renderEntry(entry) {
+
+  const $row = document.createElement('li');
+  const $columnPhoto = document.createElement('div');
+  const $columnNotes = document.createElement('div');
+  const $image = document.createElement('img');
+  const $header = document.createElement('h2');
+  const $notes = document.createElement('div');
+
+  $row.className = 'row';
+  $columnPhoto.className = 'column-full column-half';
+  $image.className = 'photo-url';
+  $image.setAttribute('alt', 'entry image');
+  $columnNotes.className = 'column-full column-half';
+  $header.className = 'title';
+  $notes.className = 'notes';
+
+  $row.appendChild($columnPhoto);
+  $columnPhoto.appendChild($image);
+  $row.appendChild($columnNotes);
+  $columnNotes.appendChild($header);
+  $columnNotes.appendChild($notes);
+
+  $image.setAttribute('src', entry.photoURL);
+  $header.textContent = entry.title;
+  $notes.textContent = entry.notes;
+
+  return $row;
+}
+
+const $list = document.querySelector('ul');
+
+document.addEventListener('DOMContentLoaded', function () {
+  for (let i = 0; i < data.entries.length; i++) {
+    $list.appendChild(renderEntry(data.entries[i]));
+  }
+  const lastView = localStorage.getItem('currentView');
+  if (lastView) {
+    viewSwap(lastView);
+  }
+  if (data.entries !== null) {
+    toggleNoEntries();
+  } else {
+    toggleNoEntries();
+  }
+});
+
+function toggleNoEntries() {
+  const $noEntry = document.querySelector('.no-entries');
+  $noEntry.className = 'no-entries-hidden';
+}
+
+function viewSwap(view) {
+  const $entriesView = document.querySelector('[data-view="entries"]');
+  const $entryFormView = document.querySelector('[data-view="entry-form"]');
```

Comment:
I would do these `querySelector`s outside of the function in this global space. By including them here, we have to traverse the DOM to find these elements every single time this function gets called. If we do it in the global space, we only need to do that search once, and we'll always have a reference to it.